### PR TITLE
chore: add setup for forked tests

### PR DESCRIPTION
**Pull request by sujithsomraaj**
related to #26

- added forked unit tests across four different chains (BSC, ETH, ARB, POLYGON)
- added pigeon helpers for (CELER & HYPERLANE)

basic tests are enough to test the new features to be added. later will add more bridges to the test suite to assert all the adapters.

### CODE REVIEW

File: test/Setup.t.sol [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        contractAddress[56][bytes("CELER_RECEIVER_ADAPTER")] = address(new CelerReceiverAdapter(BSC_CELER_MSG_BUS));
+        CelerReceiverAdapter(contractAddress[56][bytes("CELER_RECEIVER_ADAPTER")]).updateSenderAdapter(srcChain, senderAdapters);
+
+        vm.selectFork(fork[137]);
+        contractAddress[137][bytes("CELER_RECEIVER_ADAPTER")] = address(new CelerReceiverAdapter(POLYGON_CELER_MSG_BUS));
+        CelerReceiverAdapter(contractAddress[137][bytes("CELER_RECEIVER_ADAPTER")]).updateSenderAdapter(srcChain, senderAdapters);
+
+        vm.selectFork(fork[42161]);
+        contractAddress[42161][bytes("CELER_RECEIVER_ADAPTER")] = address(new CelerReceiverAdapter(ARB_CELER_MSG_BUS));
+        CelerReceiverAdapter(contractAddress[42161][bytes("CELER_RECEIVER_ADAPTER")]).updateSenderAdapter(srcChain, senderAdapters);
```
Comment (ermyas) [inline test/Setup.t.sol:156]: nit: Could this be in a `for` loop, as done above when setting the hyperplane adapters for consistency and to make it easier as new chains are added?

File: test/Setup.t.sol [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    /// @dev constants for hyperlane
+    address constant HYPERLANE_MAILBOX = 0x35231d4c2D8B8ADcB5617A638A0c4548684c7C70;
+    address constant HYPERLANE_IGP = 0x56f52c0A1ddcD557285f7CBc782D3d83096CE1Cc;
+
+    /// @dev constants for celer
+    address constant ETH_CELER_MSG_BUS = 0x4066D196A423b2b3B8B054f4F40efB47a74E200C;
+    address constant BSC_CELER_MSG_BUS = 0x95714818fdd7a5454F73Da9c777B3ee6EbAEEa6B;
+    address constant POLYGON_CELER_MSG_BUS = 0xaFDb9C40C7144022811F034EE07Ce2E110093fe6;
+    address constant ARB_CELER_MSG_BUS = 0x3Ad9d0648CDAA2426331e894e980D0a5Ed16257f;
```
[the remark this answers is not in the document]
Comment (0xkydo) [reply test/Setup.t.sol:38]: Axelar adapter is currently in its own PR https://github.com/MultiMessageAggregation/multibridge/pull/6

File: test/success-tests/BasicSendReceiveTest.t.sol [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+/// library imports
+import {Vm} from "forge-std/Test.sol";
+
+/// local imports
+import "../Setup.t.sol";
+import {MultiMessageSender} from "../../src/MultiMessageSender.sol";
+
+contract MMA is Setup {
+    /// @dev intializes the setup
```
Comment (ermyas) [inline test/success-tests/BasicSendReceiveTest.t.sol:12]: nit: "intializes" -> "initializes"

File: test/Setup.t.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xkydo) [reply test/Setup.t.sol:38]: Makes our live easier as well

**Review by 0xkydo — CHANGES_REQUESTED** [review]
All lgtm, let's adopt @ermyas 's comments and merge this pr.

**Comment by ermyas** [comment]
As discussed offline, closing this PR in favour of PRs #39 and #40

**Pull request closed without merging**